workers/unity/Assets/Playground/Scripts/EntityInitialization/GameObjectInitializationSystem.cs

```diff
@@ -32,50 +32,68 @@ public struct RemovedEntitiesData
             public SubtractiveComponent<GameObjectReference> NoGameObjectReference;
         }
 
+        public struct WorkerData
+        {
+            public readonly int Length;
+            [ReadOnly] public SharedComponentDataArray<WorkerConfig> WorkerConfigs;
+        }
+
         [Inject] private AddedEntitiesData addedEntitiesData;
         [Inject] private RemovedEntitiesData removedEntitiesData;
+        [Inject] private WorkerData workerData;
 
-        private MutableView view;
-        private WorkerBase worker;
-        private Vector3 origin;
-        private readonly ViewCommandBuffer viewCommandBuffer = new ViewCommandBuffer();
+        private ViewCommandBuffer viewCommandBuffer;
         private EntityGameObjectCreator entityGameObjectCreator;
+        private EntityGameObjectLinker entityGameObjectLinker;
         private uint currentHandle;
         private readonly Dictionary<int, GameObject> entityGameObjectCache = new Dictionary<int, GameObject>();
 
         protected override void OnCreateManager(int capacity)
         {
             base.OnCreateManager(capacity);
 
-            worker = WorkerRegistry.GetWorkerForWorld(World);
-            view = worker.View;
-            origin = worker.Origin;
             entityGameObjectCreator = new EntityGameObjectCreator(World);
         }
 
         protected override void OnUpdate()
         {
+            if (workerData.Length == 0)
+            {
+                new LoggingDispatcher().HandleLog(LogType.Error, new LogEvent("This system should not have been run without a worker entity"));
+            }
+
+            var worker = workerData.WorkerConfigs[0].Worker;
+            if (viewCommandBuffer == null)
```

Comment:
This is bad, is there no way to do this in `OnCreateManager` ?

Reply:
found out that we actually can do it. Thought we don't have access to the injected groups in `OnCreateManager`. will revert that change